File: C#/1090. Largest Values From Labels.cs
public class Solution {
    public int LargestValsFromLabels(int[] values, int[] labels, int numWanted, int useLimit) {
        int n = values.Length;
        var arr = new Pair[n];
        Dictionary<int, int> dict = new Dictionary<int, int>();
        int res = 0;
        for(int i=0; i<n; i++){
            arr[i] = new Pair(values[i], labels[i]);
        }
        Array.Sort(arr,(a,b) => b.value.CompareTo(a.value));
        for(int i=0; i<n; i++){        
            if(numWanted!=0){
                if(!dict.ContainsKey(arr[i].label)){
                    dict.Add(arr[i].label, 0);
                }
                if(dict[arr[i].label] < useLimit){   
                    dict[arr[i].label]++;
                    res+=arr[i].value;
                    numWanted--;
                }
            }
        }
        return res;
    }
    public class Pair{
        public int value;
        public int label;
        public Pair(int value, int label){
            this.value=value;
            this.label=label;
        }
    }
}

File: Java/1090. Largest Values From Labels.java
class Solution {
    public int largestValsFromLabels(int[] values, int[] labels, int numWanted, int useLimit) {
        List<Pair> l = new ArrayList<>();
        HashMap<Integer, Integer> mp = new HashMap<Integer, Integer>();
        int n = values.length;
        int res = 0;
        for(int i=0; i<n; i++){
            l.add(new Pair(values[i],labels[i]));
        }
        Collections.sort(l,(a,b)->b.value-a.value);
        for(int i=0; i<n; i++){        
            if((!mp.containsKey(l.get(i).label) || mp.get(l.get(i).label)<useLimit) && numWanted!=0){
                mp.put(l.get(i).label,mp.getOrDefault(l.get(i).label,0)+1);
                res+=l.get(i).value;
                numWanted--;
            }
        }
        return res;
    }
}
class Pair{
    int value;
    int label;
    public Pair(int value, int label){
        this.value=value;
        this.label=label;
    }
}
